File: hfdp/src/facade/Computer.rb
class CPU
    def execute; end
end

class Memory
    def load; end
end

class HardDrive
    def read; end
end

class Computer
    def initialize
        @cpu = CPU.new
        @mem = Memory.new
        @hd  = HardDrive.new
    end

    def start
        @cpu.execute
        @mem.load
        @hd.read
    end
end

computerFacade = Computer.new
computerFacade.start

File: hfdp/src/facade/Computer.java
package facade;

/**
 * Created by tisong on 1/18/17.
 */

/**
 * Computer就是外观类
 */
public class Computer {

    private CPU cpu;

    private Memory memory;

    private HardDrive hardDrive;

    /**
     * 对子系统的封装
     */
    public void startComputer() {
        cpu.execute();
        memory.load();
        hardDrive.read();
    }
}

/**
 * CPU Memory HardDrive 子系统
 */

class CPU {
    public void execute() {
        System.out.println("cpu execute");
    }
}

class Memory {
    public void load() {
        System.out.println("load memory");
    }
}

class HardDrive {
    public void read() {
        System.out.println("read hard drive");
    }
}